File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-3.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        l, r = 0, 0
        ans = 0
        count = set()
        while r < len(s):
            if s[r] not in count:
                count.add(s[r])
                ans = max(ans, (r - l) + 1)
                r += 1
            else:
                count.remove(s[l])
                l += 1

        return ans

File: Data Structures & Algorithms/max-area-of-island/submission-3.js
class Solution {
	/**
	 * @param {number[][]} grid
	 * @return {number}
	 */
	maxAreaOfIsland(grid) {
		const [ROW, COL] = [grid.length, grid[0].length]
		const dfs = (i, j) => {
			if (i >= ROW || j >= COL || i < 0 || j < 0 || grid[i][j] === 0) return 0;
			grid[i][j] = 0;
			return 1 + dfs(i + 1, j) + dfs(i - 1, j) + dfs(i, j + 1) + dfs(i, j - 1);
		}
		let ans = 0;
		for (let r = 0; r < ROW; r++) {
			for (let c = 0; c < COL; c++) {
				if (grid[r][c] === 1) {
					ans = Math.max(ans, dfs(r, c));
				}
			}
		}
		return ans;
	}
}
